lib/src/main/java/com/dynatrace/file/util/FilePollerFactory.java

```diff
@@ -0,0 +1,62 @@
+package com.dynatrace.file.util;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.logging.Logger;
+
+public class FilePollerFactory {
+  private static final Logger logger = Logger.getLogger(FilePollerFactory.class.getName());
+
+  /**
+   * See {@link FilePollerFactory#getDefault(String, Duration) getDefault} for details. It calls
+   * this method with the default poll duration of 60 seconds.
+   *
+   * @param fileName The name of the file to be watched.
+   * @return An object that implements the abstract methods on {@link AbstractFilePoller}.
+   * @throws IOException if the initialization of the {@link WatchServiceBasedFilePoller} is not
+   *     successful.
+   */
+  public static AbstractFilePoller getDefault(String fileName) throws IOException {
+    return getDefault(fileName, Duration.ofSeconds(60));
+  }
+
+  /**
+   * Creates the default {@link AbstractFilePoller} based on the current OS. For Linux and Windows
+   * based OS's, uses a {@link java.nio.file.WatchService WatchService} based approach. For macOS,
+   * where there is no native implementation for the {@link java.nio.file.WatchService WatchService}
+   * bindings this method provides a {@link PollBasedFilePoller}, which polls the file of interest
+   * periodically (according to the pollInterval).
+   *
+   * @param fileName The name of the file to be watched.
+   * @param pollInterval The interval in which the {@link PollBasedFilePoller} polls for changes.
+   *     Only applicable on macOS.
+   * @return An object that implements the abstract methods on {@link AbstractFilePoller}.
+   * @throws IOException if the initialization of the {@link WatchServiceBasedFilePoller} is not
+   *     successful.
+   */
+  public static AbstractFilePoller getDefault(String fileName, Duration pollInterval)
+      throws IOException {
+    boolean isMacOs = System.getProperty("os.name").toLowerCase().contains("mac");
```

Comment:
Since we just create one instance I think it's not a big deal, but we could save this in a static var to avoid calling it again: https://stackoverflow.com/a/228499/2689390